client/src/rpb/granular_view.js

```diff
@@ -100,70 +111,70 @@ class GranularView extends React.Component {
-      (filter_by_dim) => ({
-        ...filter_by_dim,
-        children: _.sortBy(filter_by_dim.children, "display"),
-      })
-    );
+
+    const table_data = dim_all_or_dept
+      ? _.map(flat_data, (row) => {
+          const org = Dept.store.lookup(row.dept);
+          return {
+            dept: org.name,
+            legal_title: org.legal_title ? org.legal_title : org.name,
+            ..._.chain(cols)
+              .filter((col) => _.has(row, col.nick))
+              .map(({ nick }) => [nick, row[nick]])
+              .fromPairs()
+              .value(),
+          };
+        })
+      : flat_data;
+
     const dropdown_content = (
       <div className="group_filter_dropdown">
-        {_.map(group_filter_options, (group) => (
-          <div style={{ marginBottom: 10 }} key={group.id}>
-            <span key={group.id} style={{ fontWeight: 700 }}>
-              {group.display}
-            </span>
-            {_.map(group.children, (child) => (
-              <div key={`${group.id}_${child.filter}__${child.dimension}`}>
-                <input
-                  type={"radio"}
-                  id={`${child.filter}__${child.dimension}`}
-                  name={"rpb_group_filter"}
-                  key={`${child.filter}__${child.dimension}`}
-                  onClick={() => {
-                    on_set_filter({
-                      filter: child.filter,
-                      dimension: child.dimension,
-                    });
-                  }}
-                  defaultChecked={
-                    child.filter === filter && child.dimension === dimension
-                  }
-                />
-                <label
-                  htmlFor={`${child.filter}__${child.dimension}`}
-                  className={"normal-radio-btn-label"}
-                  key={child.display}
-                >
-                  {child.display}
-                </label>
-              </div>
-            ))}
+        {_.map(dimensions, (dim) => (
+          <div style={{ marginBottom: 10 }} key={`${dim}-div`}>
+            <div>
+              <input
+                type={"radio"}
+                id={dim}
+                name={"rpb_group_filter"}
+                key={dim}
+                onClick={() => {
+                  on_set_dimension({
+                    dimension: dim,
+                  });
+                }}
+                defaultChecked={dim === dimension}
+              />
+              <label
+                htmlFor={dim}
+                className={"normal-radio-btn-label"}
+                key={`${dim}-radio-btn-label`}
+              >
+                {dim === "all" || !_.find(sorted_key_columns, ["nick", dim])
+                  ? text_maker(dim)
+                  : dim === "dept"
+                  ? text_maker("org")
+                  : _.find(sorted_key_columns, ["nick", dim]).header[lang] ||
+                    _.find(sorted_key_columns, ["nick", dim]).header["en"]}
```

Comment:
is the "en" supposed to be here?

Reply:
Doubling down on this, what is the use of the last `||` condition?

Reply:
Some of the `sorted_key_columns` only have a "en" header and some have "en" and "fr" headers

For example on Expenditures by Standard Object:
![Screen Shot 2021-11-15 at 1 51 44 PM](https://user-images.githubusercontent.com/49965519/141838036-4ce52080-f102-477d-8d33-53209c316d3b.png)

Reply:
I didn't use `fully_qualified_name` because of this:

![Screen Shot 2021-11-15 at 4 04 49 PM](https://user-images.githubusercontent.com/49965519/141853751-5c7f3e19-abdc-429e-97ad-5f7e9861929c.png)

Reply:
I used `header` instead of `fully_qualified_name` because of this (on Transfer Payments):

![Screen Shot 2021-11-15 at 4 21 23 PM](https://user-images.githubusercontent.com/49965519/141855498-cdf6b7a6-7852-4832-8f49-cf5dbeedd735.png)

With using `header`, "Transfer Payments - Type" showed up as just "Type" and I thought that looked better at the time but i could use `fully_qualified_name` instead